trans/src/Pact/Trans/Dec.hs

```diff
@@ -0,0 +1,142 @@
+{-# LANGUAGE BangPatterns #-}
+{-# LANGUAGE CPP #-}
+{-# LANGUAGE DeriveFunctor #-}
+{-# LANGUAGE DeriveFoldable #-}
+{-# LANGUAGE DeriveTraversable #-}
+
+{-# OPTIONS_GHC -Wno-incomplete-record-updates #-}
+
+-- |
+-- Module      :  Pact.Trans.Dec
+-- Copyright   :  (C) 2023 John Wiegley
+-- License     :  BSD-style (see the file LICENSE)
+-- Maintainer  :  John Wiegley <john@kadena.io>
+--
+-- Operators and math built-ins based on the MPFR library.
+--
+
+module Pact.Trans.Dec
+  ( dec_exp
+  , dec_ln
+  , dec_log
+  , dec_pow
+  , dec_sqrt
+  , dec_reduce
+  ) where
+
+import qualified Data.Decimal as D
+import Numeric.Decimal
+import qualified Numeric.Decimal.Number as N
+import Numeric.Decimal.Arithmetic
+import qualified Numeric.Decimal.Operation as Op
+import Pact.Trans.Types
+
+type Prec = P100
+
+{-
+decShow :: D.Decimal -> String
+decShow D.Decimal { D.decimalPlaces = e, D.decimalMantissa = c } =
+  "Dec { places = "
+    ++ show e ++ ", mantissa = "
+    ++ show c ++ " }"
+
+numShow :: Decimal p r -> String
+numShow N.Num { N.sign = s, N.coefficient = c, N.exponent = e } =
+  "Num { sign = "
+    ++ (case s of Op.Neg -> "Neg"; Op.Pos -> "Pos") ++ ", coefficient = "
+    ++ show c ++ ", exponent = "
+    ++ show e ++ " }"
+numShow _ = error "unexpected"
+-}
+
+toDecimal :: Decimal p r -> TransResult D.Decimal
```

Comment:
I think we should export `toDecimal` and `fromDecimal`

Reply:
Done